src/toCamelCase.ts

```diff
@@ -0,0 +1,97 @@
+import { type CamelCase } from "type-fest";
+import { splitWords } from "./internal/splitWords";
+
+const LOWER_CASE_CHARACTER_RE = /[a-z]/u;
+
+type CamelCaseOptions = {
+  readonly preserveConsecutiveUppercase?: boolean;
+};
+
+const DEFAULT_OPTIONS = {
+  // Same as type-fest's default.
+  preserveConsecutiveUppercase: true,
+} as const satisfies CamelCaseOptions;
+
+/**
+ * Convert a text to camelCase by splitting it into words, un-capitalizing the
+ * first word, capitalizing the rest, then joining them back together. This is
+ * the runtime implementation of type-fest's [`CamelCase` type](https://github.com/sindresorhus/type-fest/blob/main/source/camel-case.d.ts).
+ *
+ * For other case manipulations see: `toLowerCase`, `toUpperCase`, `capitalize`,
+ * and `uncapitalize`.
+ *
+ * For *PascalCase* use `capitalize(toCamelCase(data))`.
+ *
+ * @param data - A string.
+ * @param options - Used to disable the default behavior of preserving
+ * consecutive uppercase characters. This is optional.
+ * @signature
+ *   R.toCamelCase(data);
+ * @example
+ *   R.toCamelCase("hello world"); // "helloWorld"
```

Comment:
**question:** can we add more examples?

something like `R.toCamelCase("__HELLO_WORLD__")` or something